### PR TITLE
WASM: Enable System.Runtime.Serialization.Formatters tests

### PR DESCRIPTION
**Pull request by akoeplinger**
We need to preserve some assemblies the BinaryFormatter needs and also skip some test cases that would cause PNSE on WebAssembly: `Tests run: 6384, Errors: 0, Failures: 0, Skipped: 0. Time: 58.90189s`

### CODE REVIEW

File: src/libraries/System.Runtime.Serialization.Formatters/tests/BinaryFormatterTestData.cs [hunk 1]
```diff
@@ -681,32 +693,35 @@ public static IEnumerable<object[]> SerializableObjects()
...
 
-            SqlError sqlError = null;
-            // .NET Framework has no constructor overload for passing an exception
-            if (PlatformDetection.IsNetFramework)
+            if (!PlatformDetection.IsBrowser) // System.Data.SqlClient doesn't work on Browser
```
Comment (stephentoub) [inline src/libraries/System.Runtime.Serialization.Formatters/tests/BinaryFormatterTestData.cs:696]: @bartonjs, @joperezr, do we still want to test serialization of SqlClient types now that SqlClient is no longer in dotnet/runtime?  Separate from this PR, seems like we could probably delete this section.

**Review by stephentoub — APPROVED** [review]

File: eng/testing/tests.mobile.targets [hunk 2]
```diff
@@ -125,6 +125,20 @@
       <AssemblySearchPaths Include="$(PublishDir)"/>
       <WasmFilesToIncludeInFileSystem Include="@(ContentWithTargetPath)" />
       <WasmFilesToIncludeInFileSystem Include="@(ReferenceCopyLocalPaths)" Condition="'%(ReferenceCopyLocalPaths.BuildReference)' == 'true'" />
+      <ExtraAssemblies Include="$(PublishDir)$(AssemblyName).dll" />
+      <!-- we need to preserve these facades for BinaryFormatter tests -->
+      <ExtraAssemblies Include="$(PublishDir)mscorlib.dll" />
```
Comment (safern) [inline eng/testing/tests.mobile.targets:130]: Why not publish trimmed and remove the custom trimming we do in the AppBuilder?

File: src/libraries/tests.proj [hunk 3]
```diff
@@ -85,7 +85,6 @@
     <ProjectExclusions Include="$(MSBuildThisFileDirectory)System.Runtime.InteropServices\tests\System.Runtime.InteropServices.Tests.csproj" />
     <ProjectExclusions Include="$(MSBuildThisFileDirectory)System.Runtime.Loader\tests\DefaultContext\System.Runtime.Loader.DefaultContext.Tests.csproj" />
     <ProjectExclusions Include="$(MSBuildThisFileDirectory)System.Runtime.Loader\tests\System.Runtime.Loader.Tests.csproj" />
-    <ProjectExclusions Include="$(MSBuildThisFileDirectory)System.Runtime.Serialization.Formatters\tests\System.Runtime.Serialization.Formatters.Tests.csproj" />
```
Comment (safern) [inline src/libraries/tests.proj:88]: I believe this might get other test assemblies passing, no?

File: eng/testing/tests.mobile.targets (hunk 2 above)
Comment (akoeplinger) [reply eng/testing/tests.mobile.targets:130]: that's the plan but this was quicker for now.

File: src/libraries/tests.proj (hunk 3 above)
Comment (akoeplinger) [reply src/libraries/tests.proj:88]: we'll see once we pass through the list as part of https://github.com/dotnet/runtime/issues/38422. this was just a quick PR to unblock tests.

File: src/libraries/System.Runtime.Serialization.Formatters/tests/BinaryFormatterTestData.cs (hunk 1 above)
Comment (joperezr) [reply src/libraries/System.Runtime.Serialization.Formatters/tests/BinaryFormatterTestData.cs:696]: I'm fine with removing them given they would effectively not be catching anything any longer.

Comment (joperezr) [reply src/libraries/System.Runtime.Serialization.Formatters/tests/BinaryFormatterTestData.cs:696]: Logged https://github.com/dotnet/runtime/issues/38949 to track this.

**Review by safern — APPROVED** [review]

**Pull request merged**